### PR TITLE
Feature/3693 api components installer

### LINKED ISSUE #3693: Api-components-installer
## Is your feature request related to a problem? Please describe.

First version for "Api-components-installer" on Airy controller.

### PR DESCRIPTION
**Pull request by juan-sebastian**
closes #3693

### CODE REVIEW

File: backend/components/installer/src/main/java/co/airy/core/components/installer/CatalogHandler.java [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+        return getComponents((s -> s.equals(componentName)))
+            .stream()
+            .findAny()
+            //FIXME: throws exception instead
+            .orElse(null);
+    }
+
+    //TODO: Add return value and correct exception handleling and return value
+    private List<ComponentDetails> getComponents(Function<String, Boolean> condition) throws Exception {
+        git.pull();
```
Comment (chrismatix) [inline backend/components/installer/src/main/java/co/airy/core/components/installer/CatalogHandler.java:96]: curious: What's the reason for using this over the GitHub tree API you mentioned?

**Review by ljupcovangelski — DISMISSED** [review]
The Helm chart code looks good :+1: 

I don't fully understand why are we using a `cache`. Is it only for listing the installed components? I think the existing approach by fetching the a list of ConfigMaps worked well. It reduces the complexity (no cache) and the component is anyway useless without the ConfigMap.

File: backend/components/installer/src/main/java/co/airy/core/api/components/installer/InstallerHandlerCacheManager.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        this.helmJobHandler = helmJobHandler;
+        this.installedComponentsHandler = installedComponentsHandler;
+    }
+
+    @Async("threadPoolTaskExecutor")
+    public void resetCacheAfterJob(String jobName) {
+        try {
+            final V1Job job = helmJobHandler.getJobByName(jobName);
+            final CoreV1Api api = new CoreV1Api(apiClient);
+            helmJobHandler.waitForCompletedStatus(api, job);
```
Comment (ljupcovangelski) [inline backend/components/installer/src/main/java/co/airy/core/api/components/installer/InstallerHandlerCacheManager.java:36]: As install/uninstall can take some time, it is good that we wait for the job to be completed successfully before resetting the cache :+1:

File: backend/components/installer/src/main/java/co/airy/core/api/components/installer/InstallerHandler.java [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+            InstallerHandlerCacheManager installerHandlerCacheManager,
+            @Value("${kubernetes.namespace}") String namespace) {
+        this.apiClient = apiClient;
+        this.helmJobHandler = helmJobHandler;
+        this.catalogHandler = catalogHandler;
+        this.installerHandlerCacheManager = installerHandlerCacheManager;
+        this.namespace = namespace;
+    }
+
+    //TODO: Add return value and correct exception handleling
```
Comment (chrismatix) [inline backend/components/installer/src/main/java/co/airy/core/api/components/installer/InstallerHandler.java:56]: Same comment about implementing TODO or removing comment if no longer required

File: backend/components/installer/src/main/java/co/airy/core/api/components/installer/InstallerHandler.java [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+                configName,
+                namespace,
+                null);
+
+        //TODO: handle http error
+        final V1ConfigMap config = response.getData();
+        final Map<String, String> data = config.getData();
+        if (data == null) {
+            log.error("core-config configuration not found");
+            //TODO: do better error handleling
```
Comment (chrismatix) [inline backend/components/installer/src/main/java/co/airy/core/api/components/installer/InstallerHandler.java:132]: question: Better how? 🤔

File: backend/components/installer/src/main/java/co/airy/core/api/components/installer/model/ComponentDetails.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+public class ComponentDetails {
+
+    private final Map<String, Object> props = new HashMap<>();
+
+    @JsonAnyGetter
+    public Map<String, Object> getProps() {
+        return props;
+    }
+
+    @JsonAnySetter
```
Comment (chrismatix) [inline backend/components/installer/src/main/java/co/airy/core/api/components/installer/model/ComponentDetails.java:21]: TIL! Very cool feature

File: repositories.bzl [hunk 6]
```diff
@@ -71,6 +73,9 @@ airy_jvm_deps = [
     "io.github.openfeign:feign-core:11.8",
     "io.github.openfeign:feign-jackson:11.8",
     "io.github.openfeign:feign-okhttp:11.8",
+    "io.kubernetes:client-java:16.0.0",
+    "org.eclipse.jgit:org.eclipse.jgit:6.3.0.202209071007-r",
+    "commons-io:commons-io:2.11.0",
```
Comment (chrismatix) [inline repositories.bzl:78]: Where is the commons package being used?

**Review by chrismatix — CHANGES_REQUESTED** [review]

**Review by chrismatix — APPROVED** [review]

**Pull request merged**